bpf/lib/policy.h

```diff
@@ -180,24 +185,26 @@ __policy_check(const struct policy_entry *policy, const struct policy_entry *pol
 	if (unlikely(policy->deny))
 		return DROP_POLICY_DENY;
 
-	/* The chosen 'policy' has higher proxy port priority or if on the same it has more
+	/* The chosen 'policy' has higher precedence or if on the same precedence it has more
 	 * specific L4 match, or if also the L4 are equally specific, then the chosen policy has
 	 * an L3 match, which is considered to be more specific.
-	 * If proxy port priority is the same, then by definition either both have a proxy
+	 * If precedence is the same, then by definition either both have a proxy
 	 * redirect or neither has one, so we do not need to check if the other policy has a proxy
 	 * redirect or not.
 	 */
 	*proxy_port = policy->proxy_port;
 
 	auth_type = policy->auth_type;
 	if (unlikely(policy2 && policy2->auth_type > auth_type &&
-		     !policy->has_explicit_auth_type)) {
+		     !policy->has_explicit_auth_type &&
+		     ALLOW_PRECEDENCE(policy2) >= ALLOW_PRECEDENCE(policy))) {
```

Comment:
why do we need to mask here?

Reply:
When this function is called, 'policy2' never has the higher 'precedence', but it can be the same. Given this the masking is not needed and an equality check suffices, will change accordingly.

Reply:
Thanks for noticing, removing the masking also removed the need for the added macros :-)